server/blockchain/calypso/calypso.go

```diff
@@ -412,12 +412,7 @@ func (c calypsoCommand) createSecret(snap store.Snapshot, step execution.Step) e
 		return xerrors.Errorf("'%s' was not found among the SMCs", smcKey)
 	}
 
-	_, err := getSecret(snap, name)
-	if err != nil {
-		return xerrors.Errorf("a secret named '%s' already exists", name)
-	}
-
-	err = setSecret(snap, name, secret)
+	err := setSecret(snap, name, secret)
```

Comment:
@pierluca this fixes the test, but I am not convinced by this fix.

Reply:
why? could you elaborate? :)

Reply:
I removed a getSecret(...) with the condition on the err != nil, because we test on the error returned by the setSecret(), not the getSecret()